src/lib/gssapi/mechglue/g_init_sec_context.c

```diff
@@ -205,6 +208,19 @@ OM_uint32 *		time_rec;
 	    if (status != GSS_S_COMPLETE)
 		goto end;
 	}
+
+	if (used_req_flags == 0) {
+	    used_req_flags = union_ctx_id->req_flags;
+	    if (mech->gss_set_context_flags != NULL) {
+		status = mech->gss_set_context_flags(
```

Comment:
Not sure I follow here; in the last commit I added the create_sec_context() call. This is in a [branch](https://github.com/cipherboy/krb5/blob/aced581e1e16121f8eaa005632864922b858a332/src/lib/gssapi/mechglue/g_init_sec_context.c#L196-L224) where `union_ctx_id->internal_ctx_id == NULL`; thereafter create_sec_context() is called and if it falls, we exit. I think we should be safe from internal_ctx_id being null still when this is called. 

However, I do think it is wrong to wrap `set_context_flags()` in `used_req_flags == 0`, as that would not pass through ret_flags_understood, so I'll move the inner `if` statement outside.